### ISSUE TITLE
Document accessibility choices

### ISSUE THREAD

**Issue by TimVevida** [tailwindlabs/tailwindui-issues#342]
**For example, this components:**
- URL for category: https://tailwindui.com/components/application-ui/navigation/tabs
- Component name: all

**Describe the bug**
Choices are made in Tailwind UI with regards to accessibility, but they are not documented anywhere. Two examples:

* The tabs components show a select on small screens (mobile). My assumption is that the tab needs to be switched when the select value changes. However, performing actions on change is not good for a11y. For example, [it is prohibited by this a11y plugin](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-onchange.md). I would like to see a select with a simple button beside it to confirm the choice. For example:
![afbeelding](https://user-images.githubusercontent.com/16288300/95734454-c3ca5b80-0c83-11eb-8e73-40b7ff4241af.png)
Note: Of course the button needs some `.sr-only` text.
* A host of components use `aria-label` for screen reader texts. I would prefer to use a `<span class="sr-only">` whenever possible, because `aria-label` does not work when there is text content inside the element and because aria-labels are not translated by all browsers.

But most of all, I would like to hear from the component authors why they made certain decisions with regards to a11y.

**Additional context**
* [A discussion of aria-label and screen reader only text](https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/)

**Comment by davidluhr** [comment]
Thanks for this feedback!

To give some context, Tailwind UI is in early access, so feedback like this is really valuable for improvements. It should also be noted that the work going into https://headlessui.dev/ will also benefit Tailwind UI as the team finds more robust and accessible solutions for custom component patterns. Headless UI addresses the interactivity and functionality of components across frameworks, and that work will be a good foundation for Tailwind UI's collection of fully designed components.

1. This is a good suggestion. It makes sense that a form element without a dedicated submit button would be unexpected. How do you feel about the use of a native select on mobile? Have you found accessible and responsive tab patterns you prefer that we should consider?
1. I agree that `.sr-only` is better than `aria-label`. Translations are a great point, and I like having the text directly in the element. In recent updates, the components stick with `.sr-only`, and we're planning on refining existing components further before leaving early access.

**Comment by davidluhr** [comment]
As a related update to this, we've made several improvements to the semantic markup in the recent updates to Tailwind UI. This also includes leveraging `.sr-only` over `aria-label` where possible. The only remaining use of `aria-label` is to give an accessible name to repeated landmark elements that don't have a dedicated heading, such as multiple `<nav>` elements within the same component/page example.

If you find accessible/responsive tab patterns you would like us to consider, let us know and we can take another look at our approach to tabs on mobile.

**Comment by TimVevida** [comment]
My apologies for never replying to this - until now.

> It should also be noted that the work going into https://headlessui.dev/ will also benefit Tailwind UI as the team finds more robust and accessible solutions for custom component patterns.

I have already used headless UI for some transitions, thanks!

> How do you feel about the use of a native select on mobile? Have you found accessible and responsive tab patterns you prefer that we should consider?

A native select is great IMO, because native form inputs already have accessibility features baked in - or are supported by the screen readers. Besides, this gives a nice full screen select dropdown on most mobiles, which is easy to use for people with limited motor skills.

> [...]  leveraging `.sr-only` over `aria-label` where possible.

👍

**Comment by reinink** [comment]
Hey there @TimVevida, just following up on this issue. Thanks again for these suggestions.

As noted, we've updated Tailwind UI to use `.sr-only` wherever that makes sense, as well as `aria-label` where it makes sense.

Regarding the `<select>` navigation accessibility concerns, I can see that the a11y plugin has now [deprecated the no-onchange](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-onchange.md) rule, as this issue has been corrected, both in newer versions of browsers as well as in the DOM spec. So, from our understanding, creating navigation components in this way is now satisfactory. Of course, if you still feel a button is appropriate, you're more than welcome to add one to your apps. Tailwind UI is not prescriptive — you're welcome to make whatever changes you want.

Thanks again for this input!
